ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/ControlWrapper.java

```diff
@@ -306,7 +306,7 @@ public final void rollback () throws Unavailable, NoTransaction,
 		}
 		catch (NullPointerException ex)
 		{
-			throw new TRANSACTION_ROLLEDBACK();
+			return;
```

Comment:
But this one looks good. We tried to rollback but got NPE because there is no control (aka transaction) and we use presumed abort semantics so therefore we just return without any exception.